functions/Remove-DbaDbMailProfile.ps1

```diff
@@ -0,0 +1,125 @@
+        The target SQL Server instance or instances.
+
+    .PARAMETER SqlCredential
+        Login to the target instance using alternative credentials. Accepts PowerShell credentials (Get-Credential).
+
+        Windows Authentication, SQL Server Authentication, Active Directory - Password, and Active Directory - Integrated are all supported.
+
+        For MFA support, please use Connect-DbaInstance.
+
+    .PARAMETER Name
+        Specifies one or more database mail profile(s) to get. If unspecified, all profiles will be removed.
+
+    .PARAMETER ExcludeProfile
+        Specifies one or more database mail profile(s) to exclude.
+
+    .PARAMETER InputObject
+        Allows piping from Get-DbaDbMailProfile.
+
+    .PARAMETER WhatIf
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+        This is the default. Use -Confirm:$false to suppress these prompts.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: DatabaseMail, DBMail, Mail
+        Author: Mikey Bronowski (@MikeyBronowski), https://bronowski.it
+
+        dbatools PowerShell module (https://dbatools.io)
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Remove-DbaDbMailProfile
+
+    .EXAMPLE
+        PS C:\> Remove-DbaDbMailProfile -SqlInstance localhost, localhost\namedinstance
+
+        Removes all database mail profiles on the localhost, localhost\namedinstance instances.
+
+    .EXAMPLE
+        PS C:\> Remove-DbaDbMailProfile -SqlInstance localhost -Name MyDatabaseMailProfile
+
+        Removes MyDatabaseMailProfile database mail profile on the localhost.
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbMailProfile -SqlInstance SRV1 | Out-GridView -Title 'Select database mail profile(s) to drop' -OutputMode Multiple | Remove-DbaDbMailProfile
+
+        Using a pipeline this command gets all database mail profiles on SRV1, lets the user select those to remove and then removes the selected database mail profiles.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [Parameter(ParameterSetName = 'NonPipeline', Mandatory = $true, Position = 0)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [Parameter(ParameterSetName = 'NonPipeline')]
+        [PSCredential]$SqlCredential,
+        [Parameter(ParameterSetName = 'NonPipeline')]
+        [string[]]$Name,
```

Comment:
```suggestion
        [string[]]$Profile,
```
The filter input should match the pattern used for the Exclude parameter. `-Profile` then `-ExcludeProfile`, same matching we do with the Get command.

Reply:
It's because 
`The $PROFILE automatic variable stores the paths to the PowerShell profiles that are available in the current session.`


The New-* function uses Name .

Happy to change all of them if we are not concerned about $profile automatic variable.

Reply:
It does not effect `Get-DbDbMailProfile`.

Reply:
![image](https://user-images.githubusercontent.com/11204251/135696590-eaa9bd6f-e885-4ac1-987c-8eaad9c3414d.png)

Reply:
The command runs fine. The ScriptAnalyzer flagged that automatic variable.
Also, the example above returns the Profile under `Name` property, not `Profile`, so in this case, it would be more consistent.

Reply:
`$PROFILE` is a session level variable in PowerShell, that does not mean that variable will do the same thing with scoped to a function. Once it becomes a parameter of the function the session level variable is out of scope.

Example:

```powershell
function a ([string]$profile) {
    write-output "What is in $Profile"
}
```

Run that without providing any value and you will not get your PowerShell profile script, it will return nothing.

Reply:
> The ScriptAnalyzer flagged that automatic variable.

We ignore that rule and have no dependency on it being checked in the project.

Reply:
Thank you. I can change back to `Profile` (same for Account function), although the New-* commands will need a change as well, as they use `Name`.

Reply:
Changed it to `Profile`. @wsmelton  for the New-* commands, do we want to add alias with `Profile`, `Account` respectively or add those as parameters and convert `Name` parameter to an alias?

Reply:
I would vote to change it all to be consistent if we had it as Name bump that to alias. // @potatoqualitee

Reply:
Name as an alias? Most of our stuff uses like Database, Job, etc. Name can be an alias (seems fun) -- I know sometimes I try to use Name as a param and it doesn't exist, so yes, I endorse. For Profile, i see how that's iffy but it works 😅